### PR TITLE
Refactor structure

### LINKED ISSUE #11: Documentation is out of date - references version 0.0.2
The documentation [here](https://teaglebuilt.github.io/robotframework-aws/) is for version 0.0.2.   There are missing keywords, for example `Delete File` or `Create Session With Profile`.

### LINKED ISSUE #14: Inconsistencies with versioning
version.py reports the following:
VERSION = '0.0.2'

pypi reports:
robotframework-aws 0.0.4
Last released: Oct 4, 2019

Documentation references 0.0.2 which also causes confusion reported here: https://github.com/teaglebuilt/robotframework-AWS/issues/11

I suggest everything is synchronized. Perhaps we target a new release 0.0.5 with all current issues closed?

### LINKED ISSUE #21: Extend session configuration
I was using this library and ran into an issue when I needed to pass aws_session_token in addition to aws_access_key and aws_secret_key. Maybe it would be a good idea to add [all possible boto3 session parameters](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html) to '[Create Session With Keys](https://github.com/teaglebuilt/robotframework-aws/blob/master/src/AWSLibrary/keywords/session.py#L14)' keyword?

### LINKED ISSUE #28: Code for `Key Should Exist` (s3.py) needs to modified
The client.head_object() call needs to be inside the try block so we trap any errors and handle them appropriately.

If the key exists it will just drop through and return True.

If the key does not exist it will drop into the except block where we should check for "404" response to confirm the error is just that the key does not exist, otherwise the error is something else and should be raised.

### LINKED ISSUE #38: Keyword Documentation link is dead
Hi there,

I was looking to use S3 in my test cases, so I came across this library. When I clicked on the link provided in the `README.md` file it returns 404.

The link to documentation in `README.md` is: https://teaglebuilt.github.io/robotframework-AWS/

Please replace with a working link so users of this library can be benifited.

In case someone is looking for keyword documentation, you can generate one for yourself on your local machine by this command : `libdoc AWSLibrary  rf-aws.html`

### LINKED ISSUE #40: AttributeError: 'S3Keywords' object has no attribute 'logger'
Hi,

Thanks for the awesome project once again!
I am trying to try one of the test you have in test folder
```
*** Settings ***
Library  Collections
Library  AWSLibrary

*** Variable ***
${REGION}=  us-east-2
${BUCKET}=  infra-362-s3-nothing
${KEY}=  test_folder/test123
${PATH}=  /Users/yen/repo/devops/tools/aws/s3_MFA_unittest/test123

*** Test Case ***
Upload File
    Create Session With Keys  ${REGION}  ${ACCESS_KEY}  ${SECRET_KEY}
    Upload File  ${BUCKET}  ${KEY}  ${PATH} 
    Key Should Exist  ${BUCKET}  ${KEY}
    Delete File  ${BUCKET}  ${KEY}
    Delete All Sessions
```

but kept failing on below message 
```
=============================================================================
S3                                                                            
==============================================================================
Upload File                                                           | FAIL |
AttributeError: 'S3Keywords' object has no attribute 'logger'
------------------------------------------------------------------------------
S3                                                                    | FAIL |
1 test, 0 passed, 1 failed
==============================================================================
Output:  /Users/yen/repo/devops/tools/aws/s3_MFA_unittest/output.xml
Log:     /Users/yen/repo/devops/tools/aws/s3_MFA_unittest/log.html
Report:  /Users/yen/repo/devops/tools/aws/s3_MFA_unittest/report.html
```
the file actually get uploaded but since the code fail then robotframwork report it as fail

and I wonder you have seeing this before? if not, do you know where I should start to troubleshoot?
ᐅ python --version
Python 3.8.11

This is what I got when I turn on debug
```
(Pdb) r
==============================================================================
S3                                                                            
==============================================================================
--Return--
> /usr/local/anaconda3/envs/devops3811/lib/python3.8/site-packages/robot/running/suiterunner.py(84)start_suite()->None
-> self._executed_tests = NormalizedDict(ignore='_')
(Pdb) r
> /usr/local/anaconda3/envs/devops3811/lib/python3.8/site-packages/robot/model/visitor.py(86)visit_suite()
-> suite.setup.visit(self)
(Pdb) r
Upload File                                                           | FAIL |
AttributeError: 'S3Keywords' object has no attribute 'logger'
------------------------------------------------------------------------------
S3                                                                    | FAIL |
1 test, 0 passed, 1 failed
==============================================================================
--Return--
> /usr/local/anaconda3/envs/devops3811/lib/python3.8/site-packages/robot/model/visitor.py(90)visit_suite()->None
-> self.end_suite(suite)
(Pdb) l
 85             if self.start_suite(suite) is not False:
 86                 suite.setup.visit(self)
 87                 suite.suites.visit(self)
 88                 suite.tests.visit(self)
 89                 suite.teardown.visit(self)
 90  ->             self.end_suite(suite)
 91     
 92         def start_suite(self, suite):
 93             """Called when suite starts. Default implementation does nothing.
 94     
 95             Can return explicit ``False`` to stop visiting.
(Pdb) 
```

### LINKED ISSUE #42: Create session with aws session token
Not supported session creation with session token

### PR DESCRIPTION
**Pull request by samuelpcabral**
Refactor the library structure
Add new S3 keywords names and deprecate old ones
Add new DynamoDb keywords
Add "Create Session With Token" in Session keywords

Closes #42 
Closes #40 
Closes #38 
Closes #28 
Closes #21 
Closes #14 
Closes #11

### CODE REVIEW

File: development.md
Comment (paguilera) [inline development.md:1]: Why did you delete this one? Is it no longer applicable?

**Review by paguilera — COMMENTED** [review]
Overall, looks good. There's definitely sweeping changes in this PR. My comment about the development markdown file Is the only open question I've got at this point.

File: development.md
Comment (samuelpcabral) [reply development.md:1]: I removed it because most of the file mentions the testing part (robot and unit), and we can keep this in CONTRIBUTING.md. We also need to redo the entire testing part using localstack. I'm already doing some experiments with it...
